nevergrad/optimization/optimizerlib.py

```diff
@@ -1815,6 +1816,196 @@ def __init__(
+        if not self.buffer:
+            self.newX = []
+            self.loss = []
+            candidate = self._pcabo.ask()
+            if not isinstance(candidate, list):
+                candidate = candidate.tolist()
+            self.buffer = candidate
+        x_probe = self.buffer.pop()
+        data = np.tan(np.array(x_probe, copy=False))
+        candidate = self.parametrization.spawn_child().set_standardized_data(data)
+        candidate._meta["x_probe"] = x_probe
+        self.newX.append(candidate._meta["x_probe"])
+        return candidate
+
+    def _internal_tell_candidate(self, candidate: p.Parameter, loss: tp.FloatLoss) -> None:
+        self.loss.append(loss)
+        if not self.buffer:
+            if "x_probe" in candidate._meta:
+                self._pcabo.tell(self.newX, self.loss)
+            else:
+                data = candidate.get_standardized_data(reference=self.parametrization)
+                # Tell not asked:
+                self._pcabo.tell(np.arctan(data), loss)
+
+    def _internal_tell_not_asked(self, candidate: p.Parameter, loss: tp.FloatLoss) -> None:
+        raise errors.TellNotAskedNotSupportedError
+
+
+class ParametrizedPCABO(base.ConfiguredOptimizer):
+    """
+     Principle Component Analysis (PCA) Bayesian Optimization for dimensionality reduction in BO
+
+    References
+
+    [RaponiWB+20]
+        Raponi, Elena, Hao Wang, Mariusz Bujny, Simonetta Boria, and Carola Doerr.
+        "High dimensional bayesian optimization assisted by principal component analysis."
+        In International Conference on Parallel Problem Solving from Nature, pp. 169-183.
+        Springer, Cham, 2020.
+
+    """
+
+    no_parallelization = True
+    recast = True
+
+    # pylint: disable=unused-argument
+    def __init__(
+            self,
+            *,
+            # num_workers: int = 1,
+            init_budget: tp.Optional[int] = None,
+            n_components: tp.Optional[float] = 0.95,
+            prop_doe_factor: tp.Optional[float] = None,
+    ) -> None:
+        super().__init__(_PCABO, locals())
+
+
+# Testing the influence of n_components on the performance of PCABO
+PCABO80 = ParametrizedPCABO(n_components=0.80).set_name("PCABO80", register=True)
+PCABO90 = ParametrizedPCABO(n_components=0.90).set_name("PCABO90", register=True)
+PCABO92 = ParametrizedPCABO(n_components=0.92).set_name("PCABO92", register=True)
+PCABO95 = ParametrizedPCABO(n_components=0.95).set_name("PCABO95", register=True)
+PCABO98 = ParametrizedPCABO(n_components=0.98).set_name("PCABO98", register=True)
+
+# Testing the influence of the DoE size on the performance of PCABO
+PCABO95DoE0 = ParametrizedPCABO(n_components=0.95, prop_doe_factor=0.00).set_name("PCABO95DoE0", register=True)
+PCABO95DoE20 = ParametrizedPCABO(n_components=0.95, prop_doe_factor=0.20).set_name("PCABO95DoE20", register=True)
+PCABO95DoE50 = ParametrizedPCABO(n_components=0.95, prop_doe_factor=0.50).set_name("PCABO95DoE50", register=True)
```

Comment:
I'd rather avoid having many different options here, because the registry already contains way too many options, I'd keep the 2 or 3 most useful ones, we can decide which later.

If you need all of them for the experiments, it's easy to create them on the fly.

Reply:
Done